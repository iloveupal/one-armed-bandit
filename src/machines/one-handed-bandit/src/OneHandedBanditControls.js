import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import Button from 'Components/Button';
import ButtonRow from 'Components/ButtonRow';


export default class OneHandedBanditControls extends PureComponent {
    static propTypes = {
        onStartButtonClicked: PropTypes.func.isRequired,
        onFinishButtonClicked: PropTypes.func.isRequired,
        isSpinning: PropTypes.bool.isRequired,
    }

    render () {
        return (
            <ButtonRow>
                <Button
                    title="Start"
                    isDisabled={this.props.isSpinning}
                    onClick={this.props.onStartButtonClicked}
                />
                <Button
                    title="Finish"
                    isDisabled={!this.props.isSpinning}
                    secondary
                    onClick={this.props.onFinishButtonClicked}
                />
            </ButtonRow>
        );
    }
}
