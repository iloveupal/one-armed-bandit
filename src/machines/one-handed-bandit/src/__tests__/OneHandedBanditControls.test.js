import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import OneHandedBanditControls from '../OneHandedBanditControls';


describe('OneHandedBanditControls Component', () => {
    it('should match snapshot', () => {
        const onStartButtonClicked = jest.fn();
        const onFinishButtonClicked = jest.fn();
        
        const component = renderer.create(
            <OneHandedBanditControls
                isSpinning={false}
                onStartButtonClicked={onStartButtonClicked}
                onFinishButtonClicked={onFinishButtonClicked}
            />
        );

        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    })

    it('should pass isDisabled props to the buttons properly', () => {
        const onStartButtonClicked = jest.fn();
        const onFinishButtonClicked = jest.fn();
        
        const component = renderer.create(
            <OneHandedBanditControls
                isSpinning={false}
                onStartButtonClicked={onStartButtonClicked}
                onFinishButtonClicked={onFinishButtonClicked}
            />
        );

        const finishButton = component.root.findByProps({ title: 'Finish' });
        const startButton = component.root.findByProps({ title: 'Start' });

        expect(finishButton.props.isDisabled).toBe(true);
        expect(startButton.props.isDisabled).toBe(false);

        component.update(
            <OneHandedBanditControls
                isSpinning={true}
                onStartButtonClicked={onStartButtonClicked}
                onFinishButtonClicked={onFinishButtonClicked}
            />
        );

        expect(finishButton.props.isDisabled).toBe(false);
        expect(startButton.props.isDisabled).toBe(true);
    });

    it('should propagate the click event when it arises', () => {
        const onStartButtonClicked = jest.fn();
        const onFinishButtonClicked = jest.fn();
        
        const component = renderer.create(
            <OneHandedBanditControls
                isSpinning={false}
                onStartButtonClicked={onStartButtonClicked}
                onFinishButtonClicked={onFinishButtonClicked}
            />
        );

        const startButton = component.root.findByProps({ title: 'Start' });

        startButton.props.onClick();

        expect(onStartButtonClicked).toHaveBeenCalled();
    })
});
