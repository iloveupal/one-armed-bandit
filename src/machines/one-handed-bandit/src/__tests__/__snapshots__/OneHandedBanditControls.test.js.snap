// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OneHandedBanditControls Component should match snapshot 1`] = `
.c1 {
  width: 200px;
  height: 45px;
  border-radius: 5px;
  background-color: #2d3436;
  color: #fff;
  opacity: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row no-wrap;
  -ms-flex-flow: row no-wrap;
  flex-flow: row no-wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 2px solid #2d3436;
  margin: 0 15px;
  font-family: -apple-system,BlinkMacSystemFont,sans-serif;
  font-weight: bold;
  cursor: pointer;
}

.c2 {
  width: 200px;
  height: 45px;
  border-radius: 5px;
  background-color: #fff;
  color: #2d3436;
  opacity: 0.5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row no-wrap;
  -ms-flex-flow: row no-wrap;
  flex-flow: row no-wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 2px solid #2d3436;
  margin: 0 15px;
  font-family: -apple-system,BlinkMacSystemFont,sans-serif;
  font-weight: bold;
  cursor: pointer;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row no-wrap;
  -ms-flex-flow: row no-wrap;
  flex-flow: row no-wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 20px 0;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
  >
    <span>
      Start
    </span>
  </div>
  <div
    className="c2"
    onClick={[Function]}
  >
    <span>
      Finish
    </span>
  </div>
</div>
`;
